did/ion/operations.go

```diff
@@ -0,0 +1,324 @@
+	}
+	jsonOpBytes, err := json.Marshal(op)
+	if err != nil {
+		return errors.Wrapf(err, "marshaling anchor operation %+v", op)
+	}
+	resp, err := http.Post(strings.Join([]string{i.baseURL, "operations"}, "/"), "application/json", bytes.NewReader(jsonOpBytes))
+	if err != nil {
+		return errors.Wrapf(err, "posting anchor operation %+v", op)
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrapf(err, "could not resolve with response %+v", resp)
+	}
+	if !is2xxStatusCode(resp.StatusCode) {
+		return fmt.Errorf("anchor operation failed: %s", string(body))
+	}
+	return nil
+}
+
+type DID struct {
+	id                 string
+	suffix             string
+	longFormDID        string
+	operations         []any
+	updatePrivateKey   crypto.PrivateKeyJWK
+	recoveryPrivateKey crypto.PrivateKeyJWK
+}
+
+func (d *DID) IsEmpty() bool {
+	if d == nil {
+		return true
+	}
+	return reflect.DeepEqual(d, &DID{})
+}
+
+func (d *DID) ID() string {
+	return d.id
+}
+
+func (d *DID) LongForm() string {
+	return d.longFormDID
+}
+
+func (d *DID) Operations() []any {
+	return d.operations
+}
+
+func (d *DID) Operation(index int) any {
+	return d.operations[index]
+}
+
+// NewIONDID creates a new ION DID with a new recovery and update key pair in addition to any content
+// passed into in the document parameter. The document parameter is optional and can be nil.
+// The result is a DID object that contains the long form DID, and operations to be submitted to an anchor service.
+func NewIONDID(doc Document) (*DID, *CreateRequest, error) {
+	if doc.IsEmpty() {
+		return nil, nil, errors.New("document is empty")
+	}
+
+	// generate update key pair
+	_, updatePrivateKey, err := crypto.GenerateSECP256k1Key()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "generating update keypair")
+	}
+	updatePubKeyJWK, updatePrivKeyJWK, err := crypto.PrivateKeyToPrivateKeyJWK(updatePrivateKey)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "converting update key pair to JWK")
+	}
+
+	// generate recovery key pair
+	_, recoveryPrivateKey, err := crypto.GenerateSECP256k1Key()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "generating recovery keypair")
+	}
+	recoveryPubKeyJWK, recoveryPrivKeyJWK, err := crypto.PrivateKeyToPrivateKeyJWK(recoveryPrivateKey)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "converting recovery keypair to JWK")
+	}
+
+	createRequest, err := NewCreateRequest(*recoveryPubKeyJWK, *updatePubKeyJWK, doc)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "generating create request")
+	}
+
+	longFormDID, err := CreateLongFormDID(*recoveryPubKeyJWK, *updatePubKeyJWK, doc)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "generating long form DID")
+	}
+	shortFormDID, err := GetShortFormDIDFromLongFormDID(longFormDID)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "generating short form DID")
+	}
+	suffix := strings.Split(shortFormDID, ":")[2]
```

Comment:
No need to split.
```go
	suffix := shortFormDID[len("did:ion:"):]
```

But if you really want to be paranoid, then I would recommend `strings.CutPrefix` instead.